packages/apps/reputation-oracle/server/src/modules/kyc/kyc.service.ts

```diff
@@ -94,6 +92,27 @@ export class KycService {
     kycEntity.status = KycStatus.NONE;
     kycEntity.userId = userEntity.id;
 
+    const sessionInfo = await firstValueFrom(
+      this.httpService
+        .get(
+          `/individual/session/${data.session_id}/step/${this.synapsConfigService.documentID}`,
+          {
+            baseURL: this.synapsConfigService.baseUrl,
+            headers: { 'Api-Key': this.synapsConfigService.apiKey },
+          },
+        )
+        .pipe(map((response) => response.data)),
+    );
+
+    if (
+      sessionInfo?.document?.country &&
+      sessionInfo.document.country.trim() !== ''
+    ) {
+      kycEntity.country = countriesA3ToA2[sessionInfo.document.country];
+    } else {
+      throw new ControlledError(ErrorKyc.CountryNotSet, HttpStatus.BAD_REQUEST);
+    }
+
```

Comment:
This should not be here. This is the init process, Synaps doesn't have this kind of data in this step

Reply:
I think that workers do not pass identity verification using reputation oracle, they do it through HUMAN APP, correct? As far as I understand, at the moment when they initiate a session via email, they must already have passed verification.

Reply:
Reputation Oracle is like the backend of Human App, at initiate session there is no info about the user in synaps, it is the first step

Reply:
Okay, got you, working on improvements

Reply:
This logic was updated